ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2

```diff
@@ -550,6 +550,37 @@ interface {{ ethernet_interface.name }}
 {%     if ethernet_interface.poe.legacy_detect is arista.avd.defined(true) %}
    poe legacy detect
 {%     endif %}
+{%     if ethernet_interface.switchport.port_security is arista.avd.defined %}
+{%         if ethernet_interface.switchport.port_security.enabled is arista.avd.defined(true) or ethernet_interface.switchport.port_security.violation.mode is arista.avd.defined('shutdown') %}
+   switchport port-security
+{%         elif ethernet_interface.switchport.port_security.violation.mode is arista.avd.defined('protect') %}
+{%             if ethernet_interface.switchport.port_security.violation.protect_log is arista.avd.defined(true) %}
+   switchport port-security violation protect log
+{%             else %}
+   switchport port-security violation protect
+{%             endif %}
+{%         endif %}
+{%         if ethernet_interface.switchport.port_security.mac_address_maximum.disabled is arista.avd.defined(true) %}
+   switchport port-security mac-address maximum disabled
+{%         elif ethernet_interface.switchport.port_security.mac_address_maximum.disabled is arista.avd.defined(false) %}
+   no switchport port-security mac-address maximum disabled
+{%         elif ethernet_interface.switchport.port_security.mac_address_maximum.limit is arista.avd.defined %}
+   switchport port-security mac-address maximum {{ ethernet_interface.switchport.port_security.mac_address_maximum.limit }}
+{%         endif %}
+{%         if ethernet_interface.switchport.port_security.vlans is arista.avd.defined and ethernet_interface.switchport.port_security.violation.mode is not arista.avd.defined('protect') %}
+{%             for vlan in ethernet_interface.switchport.port_security.vlans %}
+{%                 if vlan.range is arista.avd.defined and vlan.mac_address_maximum is arista.avd.defined %}
+{%                     for id in vlan.range | arista.avd.range_expand %}
+{%                         set vlan_id = 'vlan ' ~ id %}
+   switchport port-security {{ vlan_id }} mac-address maximum {{ vlan.mac_address_maximum }}
+{%                     endfor %}
+{%                 endif %}
+{%             endfor %}
+{%         endif %}
+{%         if ethernet_interface.switchport.port_security.vlan_default_mac_address_maximum is arista.avd.defined and ethernet_interface.switchport.port_security.violation.mode is not arista.avd.defined('protect') %}
```

Comment:
Since the second part of this is the same check as you do in line 570 maybe you could include them in a single outer `if`

Reply:
Fixed